server/consumer.go

```diff
@@ -1830,23 +1830,73 @@ func (o *Consumer) checkPending() {
 
 // SeqFromReply will extract a sequence number from a reply subject.
 func (o *Consumer) SeqFromReply(reply string) uint64 {
-	_, seq, _, _, _ := o.ReplyInfo(reply)
-	return seq
+	_, dseq, _ := ackReplyInfo(reply)
+	return dseq
 }
 
 // StreamSeqFromReply will extract the stream sequence from the reply subject.
 func (o *Consumer) StreamSeqFromReply(reply string) uint64 {
-	seq, _, _, _, _ := o.ReplyInfo(reply)
-	return seq
+	sseq, _, _ := ackReplyInfo(reply)
+	return sseq
 }
 
+// Quick parser for positive numbers in ack reply encoding.
+func parseAckReplyNum(d string) (n int64) {
```

Comment:
We have parseInt64() already in util. You would still have to pass `[]byte(d)` instead.

Reply:
Yes and that can cause a copy..